File: app/models/location.rb
# == Schema Information
#
# Table name: locations
#
#  id          :bigint(8)        not null, primary key
#  description :text
#  latitude    :float
#  longitude   :float
#  name        :string
#  created_at  :datetime         not null
#  updated_at  :datetime         not null
#

class Location < ApplicationRecord
  has_one_attached :image
end

File: app/javascript/components/Locations/Location.js
import React from 'react';
import PropTypes from 'prop-types';

const Location = ({ location }) => {
  return (
    <div className="card">
      <div className="card-title">{location.name}</div>
      <div className="card-actions">
        <a href={`/locations/${location.id}`}>show</a>
        <a href={`/locations/${location.id}/edit`}>edit</a>
        <a
          data-method="delete"
          href={`/locations/${location.id}`}
          rel="nofollow"
        >
          delete
        </a>
      </div>
    </div>
  );
};

Location.propTypes = {
  location: PropTypes.shape({
    created_at: PropTypes.string,
    description: PropTypes.string,
    id: PropTypes.number,
    name: PropTypes.string,
    latitude: PropTypes.number,
    longitude: PropTypes.number,
    updated_at: PropTypes.string,
  }),
};

export default Location;
